From b251d68ea84ad0e31e0dfb0871169d94d9785197 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Mon, 25 Nov 2019 15:16:30 +0100
Subject: [PATCH] PONRTSYS-5378: Remove redundant code

Instead of having two a little bit different calls to
dp_register_subif_ext() just use one call and change the arguments
depending on the setting.

This should not change any functionality.
---
 drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c | 47 +++++++++---------------
 1 file changed, 17 insertions(+), 30 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
index 079ab990c6c6..bac61736e235 100644
--- a/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
+++ b/drivers/net/ethernet/lantiq/ltq_eth_drv_xrx500.c
@@ -529,22 +529,15 @@ static void ltq_eth_uninit(struct net_device *dev)
 		return;
 	}
 
-	if (priv->lct_en == 1) {
-		priv->dp_subif.subif = -1;
-		priv->dp_subif.port_id = priv->dp_port_id;
-		data.flag_ops = DP_F_DATA_LCT_SUBIF;
-		ret = dp_register_subif_ext(0, priv->owner,
-					    dev, dev->name,
-					    &priv->dp_subif,
-					    &data, DP_F_DEREGISTER);
-	} else {
-		priv->dp_subif.subif = -1;
-		priv->dp_subif.port_id = priv->dp_port_id;
-		ret = dp_register_subif_ext(0, priv->owner,
-					    dev, dev->name,
-					    &priv->dp_subif,
-					    NULL, DP_F_DEREGISTER);
-	}
+	if (priv->lct_en == 1)
+		data.flag_ops |= DP_F_DATA_LCT_SUBIF;
+
+	priv->dp_subif.subif = -1;
+	priv->dp_subif.port_id = priv->dp_port_id;
+	ret = dp_register_subif_ext(0, priv->owner,
+				    dev, dev->name,
+				    &priv->dp_subif,
+				    &data, DP_F_DEREGISTER);
 	if (ret != DP_SUCCESS)
 		pr_err("%s: failed to close for device: %s ret %d\n",
 		       __func__, dev->name, ret);
@@ -960,20 +953,14 @@ static int ltq_eth_init(struct net_device *dev)
 			dev->ethtool_ops = &serdes_ethtool_ops;
 	}
 
-	if (priv->lct_en == 1) {
-		priv->dp_subif.subif = -1;
-		priv->dp_subif.port_id = priv->dp_port_id;
-		data.flag_ops = DP_F_DATA_LCT_SUBIF;
-		ret = dp_register_subif_ext(0, priv->owner,
-					    dev, dev->name, &priv->dp_subif,
-					    &data, 0);
-	} else {
-		priv->dp_subif.subif = -1;
-		priv->dp_subif.port_id = priv->dp_port_id;
-		ret = dp_register_subif_ext(0, priv->owner,
-					    dev, dev->name, &priv->dp_subif,
-					    NULL, 0);
-	}
+	if (priv->lct_en == 1)
+		data.flag_ops |= DP_F_DATA_LCT_SUBIF;
+
+	priv->dp_subif.subif = -1;
+	priv->dp_subif.port_id = priv->dp_port_id;
+	ret = dp_register_subif_ext(0, priv->owner,
+				    dev, dev->name, &priv->dp_subif,
+				    &data, 0);
 	if (ret != DP_SUCCESS) {
 		pr_err("%s: failed to open for device: %s ret %d\n",
 		       __func__, dev->name, ret);
